frontend/src/concepts/modelRegistry/hooks/useModelVersionTuningData.ts

```diff
@@ -0,0 +1,57 @@
+import { useContext } from 'react';
+import type { ModelVersion } from '~/concepts/modelRegistry/types';
+import useModelArtifactsByVersionId from '~/concepts/modelRegistry/apiHooks/useModelArtifactsByVersionId';
+import { getDisplayNameFromK8sResource } from '~/concepts/k8s/utils';
+import { getServerAddress } from '~/pages/modelRegistry/screens/utils';
+import { ModelRegistrySelectorContext } from '~/concepts/modelRegistry/context/ModelRegistrySelectorContext';
+
+export const useModelVersionTuningData = (
+  modelVersionId: string | null,
+  modelVersion: ModelVersion | null,
+): {
+  tuningData: {
+    modelRegistryName: string;
+    modelRegistryDisplayName: string;
+    registeredModelId: string;
+    registeredModelName: string;
+    modelVersionId: string;
+    modelVersionName: string;
+    inputModelLocationUri: string;
+    outputModelRegistryApiUrl: string;
+  } | null;
+  loaded: boolean;
+  loadError: Error | null;
+} => {
+  const { preferredModelRegistry, modelRegistryServices } = useContext(
+    ModelRegistrySelectorContext,
+  );
+  const registryService = modelRegistryServices.find(
+    (s) => s.metadata.name === preferredModelRegistry?.metadata.name,
+  );
+
+  const [artifacts, loaded, loadError] = useModelArtifactsByVersionId(modelVersionId || undefined);
```

Comment:
Hmm. This is fine for now, but this means on the version details page we are fetching the artifacts twice (they also get fetched in ModelVersionDetailsView). We probably eventually want to refactor so these are lifted out to the page level or the context level. I'll make a note to follow up with a task for that, but this works fine for now.